contracts/schemes/DxDaoSchemeConstraints.sol

```diff
@@ -101,7 +105,7 @@ contract DxDaoSchemeConstraints is SchemeConstraints {
         "invalid length _tokensPeriodLimits");
         for (uint i = 0; i < _tokensAddresses.length; i++) {
             periodLimitToken[_tokensAddresses[i]] = _tokensPeriodLimits[i];
-            emit UpdatedPeriodLimitsTokens(_tokensAddresses[i], _tokensPeriodLimits[i]);
+            emit UpdatedPeriodLimitsTokens(_tokensAddresses, _tokensPeriodLimits);
```

Comment:
this should be out of the loop.
you should have a test which check that only a single event is emitted here